spec/unleash/configuration_spec.rb

```diff
@@ -107,23 +107,35 @@
       Unleash.configure do |config|
         config.url      = 'http://test-url/'
         config.app_name = 'my-test-app'
-        config.custom_http_headers = { 'X-API-KEY': '123' }
+        config.custom_http_headers = { 'X-API-KEY' => '123' }
       end
       expect{ Unleash.configuration.validate! }.not_to raise_error
-      expect(Unleash.configuration.custom_http_headers).to eq({ 'X-API-KEY': '123' })
+      expect(Unleash.configuration.custom_http_headers).to eq({ 'X-API-KEY' => '123' })
+      expect(Unleash.configuration.http_headers).to include({ 'X-API-KEY' => '123' })
     end
 
     it "should allow hashes for custom_http_headers via new client" do
       config = Unleash::Configuration.new(
         url: 'https://testurl/api',
         app_name: 'test-app',
-        custom_http_headers: { 'X-API-KEY': '123' }
+        custom_http_headers: { 'X-API-KEY' => '123' }
       )
 
       expect{ config.validate! }.not_to raise_error
-      expect(config.custom_http_headers).to include({ 'X-API-KEY': '123' })
-      expect(config.http_headers).to include({ 'UNLEASH-APPNAME' => 'test-app' })
-      expect(config.http_headers).to include('UNLEASH-INSTANCEID')
+      expect(config.custom_http_headers).to eq({ 'X-API-KEY' => '123' })
+      expect(config.http_headers).to eq('X-API-KEY' => '123', 'UNLEASH-APPNAME' => 'test-app', 'UNLEASH-INSTANCEID' => config.instance_id)
```

Comment:
(Bike-shedding) For readability would prefer to have to have the hash broken into multiple lines:
```suggestion
      expect(config.http_headers).to eq({
        'X-API-KEY' => '123',
        'UNLEASH-APPNAME' => 'test-app',
        'UNLEASH-INSTANCEID' => config.instance_id
       })
```